packages/pigeon/lib/swift_generator.dart

```diff
@@ -627,7 +627,8 @@ import FlutterMacOS
       if (listEncodedClassNames != null &&
           listEncodedClassNames.contains(type.baseName)) {
         indent.writeln('var $variableName: $fieldType? = nil');
-        indent.write('if let ${variableName}List = $value as! [Any?]? ');
+        indent
+            .write('if let ${variableName}List: [Any?] = nilOrValue($value) ');
```

Comment:
If it's `nilOrValue`, doesn't the type need to still be `[Any?]?`?

Reply:
oh it's just `[Any?]` because it's in an `if let` statement. For example: 

```
if let foo: String = anOptionalString {}
```